configuration/azure/appconfig/appconfig.go

```diff
@@ -248,10 +247,8 @@ func (r *ConfigurationStore) getAll(ctx context.Context, req *configuration.GetR
 		},
 		nil)
 
-	// TODO: here contxt.TODO() is used because the SDK panics when a cancelled context is passed in NextPage
-	// Issue - https://github.com/Azure/azure-sdk-for-go/issues/19223 . It needs to be modified to use ctx once the SDK is fixed
 	for allSettingsPgr.More() {
-		if revResp, err := allSettingsPgr.NextPage(context.TODO()); err == nil {
+		if revResp, err := allSettingsPgr.NextPage(timeoutContext); err == nil {
```

Comment:
I think you need a new `timeoutCtx` for each call, not one for the method. This is because each `NextPage` invocation makes a network call, so if you have a single timeout, getting a page after the first may fail.